lib/ddtrace/configuration/validation_proxy.rb

```diff
@@ -0,0 +1,44 @@
+require 'forwardable'
+
+module Datadog
+  module Configuration
+    # Forwards configuration settings allowed for tracing,
+    # but raises errors for access to anything else.
+    class TracingValidationProxy
+      extend Forwardable
+
+      FORWARDED_METHODS = [
+        :analytics,
+        :api_key,
+        :diagnostics,
+        :distributed_tracing,
+        :env,
+        :log_injection,
+        :logger,
+        :report_hostname,
+        :runtime_metrics,
+        :runtime_metrics_enabled,
+        :sampling,
+        :service,
+        :site,
+        :tags,
+        :test_mode,
+        :time_now_provider,
+        :tracer,
+        :version
```

Comment:
`env`/`service`/`version`/`tags`/`logger` actually raise a good question: since they're global, how can we make that clear to customers?

E.g. right now there's a bit of a mix here between tracing-only and global configurations, and it's a bit unclear to customers which ones are which.

For instance, a customer could reasonably do

```ruby
Datadog::Tracing.configure { |c| c.logger = TRACING_LOGGER }
Datadog::Profiling.configure { |c| c.logger = PROFILING_LOGGER }
```

...and discover that it actually doesn't do what they might think it does.

Same for some of the other things (for instance, only enabling diagnostics for a single component?), etc.

Reply:
I think its a fair point, but reasonable behavior under the current circumstances (given its no change in behavior or clarity.) Right now users who focus on tracing can and will do `Datadog.configure { |c| c.logger }` thinking they're only configuring the tracer logger when they're configuring the logger for everything.

This change opens the door to allowing specific loggers for each in 1.x without introducing breaking changes. So in the interim, I think it would be reasonable to update the documentation to clarify this behavior as it exists today, then actually change it later.